net/ping/ping.go

```diff
@@ -38,7 +42,15 @@ type outstanding struct {
 // A new instance should be created for each concurrent set of ping requests;
 // this type should not be reused.
 type Pinger struct {
-	c       net.PacketConn
+	// ListenPacketForTesting is used to create the underlying net.PacketConn.
+	// If nil, netns.Listener is used.
+	// This only exists for testing.
+	ListenPacketForTesting func(ctx context.Context, typ, addr string) (net.PacketConn, error)
```

Comment:
We could decouple this package from `netns` and from `netmon` if we do this:

```
type ListenPacketer interface {
  ListenPacket(context.Context, string, string) (PacketConn, error)
}

func New(ctx context.Context, logf logger.Logf, lp ListenPacketer) ...
```

It moves a little burden in to the caller, but now they can pick from `net.ListenConfig`, `netns.Listener` and so on, and unit tests can likely avoid both netns and netmon entirely.